### PR TITLE
mypy fix helprers, initiator, linter

### PR DESCRIPTION
**Pull request by jochman**
## Status
- [x] Ready

image

fixes:
```python
demisto_sdk/commands/lint/helpers.py:100: error: Incompatible return value type (got "int", expected "bool")
demisto_sdk/commands/lint/helpers.py:266: error: Name 'container_obj' already defined on line 260
demisto_sdk/commands/lint/helpers.py:289: error: Incompatible return value type (got "str", expected "float")
demisto_sdk/commands/lint/helpers.py:309: error: Item "None" of "Optional[IO[bytes]]" has no attribute "read"
demisto_sdk/commands/lint/helpers.py:313: error: Incompatible return value type (got "Union[bytes, Any]", expected "str")
demisto_sdk/commands/init/initiator.py:258: error: Incompatible types in assignment (expression has type "int", variable has type "str")
demisto_sdk/commands/init/initiator.py:267: error: Unsupported operand types for - ("str" and "int")
```

### CODE REVIEW

File: demisto_sdk/commands/lint/helpers.py [hunk 1]
```diff
@@ -257,58 +259,64 @@ def get_python_version_from_image(image: str) -> float:
         float: Python version X.Y (3.7, 3.6, ..)
     """
     docker_client = docker.from_env()
-    container_obj: Container = None
-    py_num = ""
-    for trial1 in range(2):
+    containers = list()
```
Comment (Itay4) [inline demisto_sdk/commands/lint/helpers.py:262]: why do we need that list ?

File: demisto_sdk/commands/lint/helpers.py [hunk 2]
```diff
@@ -257,58 +259,64 @@ def get_python_version_from_image(image: str) -> float:
...
             # Wait for container to finish
             container_obj.wait(condition="exited")
             # Get python version
             py_num = container_obj.logs()
             if isinstance(py_num, bytes):
                 py_num = float(py_num)
-                break
             else:
                 raise docker.errors.ContainerError
+            for _ in range(2):
```
Comment (Itay4) [inline demisto_sdk/commands/lint/helpers.py:283]: why did you move it here?

File: demisto_sdk/commands/lint/helpers.py [hunk 3]
```diff
@@ -257,58 +259,64 @@ def get_python_version_from_image(image: str) -> float:
...
     """
+    data: Union[str, bytes] = b''
     archive, stat = container_obj.get_archive(container_path)
-    filelike = io.BytesIO(b"".join(b for b in archive))
-    tar = tarfile.open(fileobj=filelike)
-    data = tar.extractfile(stat['name']).read()
-    if encoding:
-        data = data.decode(encoding)
+    file_like = io.BytesIO(b"".join(b for b in archive))
+    tar: Optional[tarfile.TarFile] = tarfile.open(fileobj=file_like)
```
Comment (Itay4) [inline demisto_sdk/commands/lint/helpers.py:313]:
```suggestion
    tar: tarfile.TarFile = tarfile.open(fileobj=file_like)
```

File: demisto_sdk/commands/lint/helpers.py [hunk 4]
```diff
@@ -257,58 +259,64 @@ def get_python_version_from_image(image: str) -> float:
...
+    if isinstance(tar, tarfile.TarFile):
```
Comment (Itay4) [inline demisto_sdk/commands/lint/helpers.py:314]: remove that condition

File: demisto_sdk/commands/lint/helpers.py [hunk 5]
```diff
@@ -257,58 +259,64 @@ def get_python_version_from_image(image: str) -> float:
...
     """ Copy file from container.
 
     Args:
         container_obj(Container): Container ID to copy file from
         container_path(Path): Path in container image (file)
-        encoding(str): valide encoding e.g. utf-8
+        encoding(str): valid encoding e.g. utf-8
 
     Returns:
         str: file as string decode as utf-8
```
Comment (Itay4) [inline demisto_sdk/commands/lint/helpers.py:305]:
```suggestion
        str or bytes: File as string decoded in utf-8
```

File: demisto_sdk/commands/lint/helpers.py (hunk 2 above)
Comment (jochman) [reply demisto_sdk/commands/lint/helpers.py:283]: because before that the code tried to use container_obj which can be not assigned

File: demisto_sdk/commands/lint/helpers.py (hunk 3 above)
Comment (jochman) [reply demisto_sdk/commands/lint/helpers.py:313]: it's Optional, mypy will fail without the optional annotation

File: demisto_sdk/commands/lint/helpers.py (hunk 4 above)
Comment (jochman) [reply demisto_sdk/commands/lint/helpers.py:314]: it must be because tafile.open can return None
